all_test.go

```diff
@@ -1163,33 +1164,135 @@ func TestDecode_formattest(t *testing.T) {
+	Expect(t, len(root.Outputs)).ToBe(1)
+	Expect(t, root.Outputs[0].ID).ToBe("output")
+	Expect(t, root.Outputs[0].Types[0].Type).ToBe("File")
+	Expect(t, root.Outputs[0].Binding.Glob).ToBe("output.txt")
+	Expect(t, root.Outputs[0].Format).ToBe("$(inputs.input.format)")
+	Expect(t, len(root.BaseCommands)).ToBe(1)
+	Expect(t, root.BaseCommands[0]).ToBe("rev")
+	Expect(t, root.Stdout).ToBe("output.txt")
+}
+
+func TestDecode_glob_expr_list(t *testing.T) {
+	f := cwl("glob-expr-list.cwl")
+	root := NewCWL()
+	Expect(t, root).TypeOf("*cwl.Root")
+	err := root.Decode(f)
+	Expect(t, err).ToBe(nil)
+	Expect(t, root.Version).ToBe("v1.0")
+	Expect(t, root.Class).ToBe("CommandLineTool")
+	Expect(t, len(root.Inputs)).ToBe(1)
+	Expect(t, root.Inputs[0].ID).ToBe("ids")
+	Expect(t, root.Inputs[0].Types[0].Type).ToBe("string[]")
+	Expect(t, root.Inputs[0].Binding.Position).ToBe(1)
+	Expect(t, len(root.Outputs)).ToBe(1)
+	Expect(t, root.Outputs[0].ID).ToBe("files")
+	Expect(t, root.Outputs[0].Types[0].Type).ToBe("File[]")
+	Expect(t, root.Outputs[0].Binding.Glob).ToBe("$(inputs.ids)")
+	Expect(t, len(root.BaseCommands)).ToBe(1)
+	Expect(t, root.BaseCommands[0]).ToBe("touch")
+}
+
+func TestDecode_imported_hint(t *testing.T) {
+	f := cwl("imported-hint.cwl")
+	root := NewCWL()
+	Expect(t, root).TypeOf("*cwl.Root")
+	err := root.Decode(f)
+	Expect(t, err).ToBe(nil)
+	Expect(t, root.Version).ToBe("v1.0")
+	Expect(t, root.Class).ToBe("CommandLineTool")
+	Expect(t, len(root.Inputs)).ToBe(0)
+	Expect(t, len(root.Outputs)).ToBe(1)
+	Expect(t, root.Outputs[0].ID).ToBe("out")
+	// TODO test out: stdout 's stdout
+	Expect(t, root.Hints).TypeOf("cwl.Hints")
+	Expect(t, root.Hints[0].Import).ToBe("envvar.yml")
+	Expect(t, len(root.BaseCommands)).ToBe(3)
+	Expect(t, root.BaseCommands[0]).ToBe("/bin/bash")
+	Expect(t, root.BaseCommands[1]).ToBe("-c")
+	Expect(t, root.BaseCommands[2]).ToBe("echo $TEST_ENV")
+	Expect(t, root.Stdout).ToBe("out")
+}
+
+func TestDecode_initialwork_path(t *testing.T) {
+	f := cwl("initialwork-path.cwl")
+	root := NewCWL()
+	Expect(t, root).TypeOf("*cwl.Root")
+	err := root.Decode(f)
+	Expect(t, err).ToBe(nil)
+	Expect(t, root.Version).ToBe("v1.0")
+	Expect(t, root.Class).ToBe("CommandLineTool")
+	Expect(t, len(root.Inputs)).ToBe(1)
+	Expect(t, root.Inputs[0].ID).ToBe("file1")
+	Expect(t, root.Inputs[0].Types[0].Type).ToBe("File")
+	Expect(t, len(root.Outputs)).ToBe(0)
+	Expect(t, root.Requirements[0].Class).ToBe("InitialWorkDirRequirement")
+	Expect(t, root.Requirements[0].Listing[0].Name).ToBe("bob.txt")
+	Expect(t, root.Requirements[0].Listing[0].Entry).ToBe(`$(inputs.file1)`)
+	Expect(t, root.Requirements[1].Class).ToBe("ShellCommandRequirement")
+	fmt.Println(root.Arguments[0])
+	a := root.Arguments[0].CommandLineBinding
+	Expect(t, a["shellQuote"]).ToBe(false)
+	str := a["valueFrom"].(string)[0 : utf8.RuneCountInString(a["valueFrom"].(string))-1]
+	Expect(t, str).ToBe("test \"$(inputs.file1.path)\" = \"$(runtime.outdir)/bob.txt\"")
```

Comment:
I guess it can be written like this https://github.com/otiai10/cwl.go/blob/master/all_test.go#L814-L816

by using 

```
s := `chars
with
new
line
`
```